### DISCUSSION TITLE
Clarify if "should-total" duration allowed to be negative

### DISCUSSION THREAD

**Discussion by vladdeSV** [jotaen/klog#388] [Something else]
> A should-total MUST be a duration value [...]

It feels unnatural to specify a negative duration, however the spec allows it.

I believe it would be good to clarify this by either
- giving a negative duration as an example, or
- specifying only positive durations are allowed (what I assumed to be the case)

**Comment by jotaen** [comment]
Yeah, good question. It’s allowed intentionally, because there is no *technical* reason to reject it. I thought maybe someone has a use-case like “my manager told me to take half a day off because I did such a great job yesterday” and wants to note that down as a separate record rather than deducting it somewhere.

It’s a bit artificial and maybe not ever needed, but on the other hand it shouldn’t do any harm to allow it. (People don’t *have* to use it, after all.) I agree, though, that it’s somewhat unintuitive and that the spec could make this clearer! Probably best to add an extra sentence à la `A negative duration MAY be used.`

**Comment by vladdeSV** [comment]
I think that would be a great addition to the spec 👍

**Comment by jotaen** [comment]
Done https://github.com/jotaen/klog/commit/8927d2e4df204bccac5cf3cbec056530e0c1b262
